pkg/build/api/v1/types.go

```diff
@@ -163,6 +163,9 @@ type BuildSource struct {
 	// repository.
 	ContextDir string `json:"contextDir,omitempty" description:"specifies sub-directory where the source code for the application exists, allows for sources to be built from a directory other than the root of a repository"`
 
+	// DockerfileName is the name of the Dockerfile that will be used to build the Docker image.
+	DockerfileName string `json:"dockerfileName,omitempty" description:"specifies the name of the Dockerfile to use for building the Docker image"`
```

Comment:
@bparees but going down your path we end up blowing up our BuildSource structure with data that are applicable only for certain strategies. That's why I always liked the strategy approach, where you define strategy-specific fields. 

Btw. I've wondered about the existence of Dockerfile string field in BuildSource, what it's for in the first place? If I'm looking correctly it's for [overwriting source's Dockerfile](https://github.com/openshift/origin/blob/e66f133f88238b55441bab4772ed2456c72c7f01/pkg/build/builder/source.go#L60), is that correct? What for? Isn't source repo enough for that? If so, shouldn't if be part of SourceBuildStrategy, especially that in [here](https://github.com/openshift/origin/blob/e66f133f88238b55441bab4772ed2456c72c7f01/pkg/build/generator/generator.go#L375) we are clearing that field under certain conditions, which even more tells me it's not right.
Sorry for those lame questions, but I wasn't following builds for a while and I'm kinda confused about those changes.